### PR TITLE
Fix the request permission for applications using > titanium 5.1.0

### PR DESCRIPTION
**Pull request by williamrijksen**
Fixes the problem on iOS when by requesting permissions on > Ti 5.1.0.

See: http://docs.appcelerator.com/platform/latest/#!/api/Titanium.Geolocation-method-hasLocationPermissions.
and: 
http://docs.appcelerator.com/platform/latest/#!/api/Titanium.Geolocation-method-requestLocationPermissions

### CODE REVIEW

File: lib/scanners/base.js [hunk 1]
```diff
@@ -34,17 +34,41 @@ var BaseScanner = function (beaconMapper, beaconRegionMapper, beaconRegionMonito
...
-            }
-
-            beaconHandler.init();
-            beaconLog.init();
-            self.setBeaconRegions([]);
-            if (OS_ANDROID || (OS_IOS && (_.isUndefined(Ti.App.arguments.launchOptionsLocationKey) || Ti.App.arguments.launchOptionsLocationKey === false))) {
-                knownBeaconService.refreshBeacons([self.networkId]);
+                var version = Ti.version.split(".");
+                var permissionType = Ti.Geolocation.AUTHORIZATION_ALWAYS;
+                if (version[0] < 5 || // Version < 5
```
Comment (david4worx) [inline lib/scanners/base.js]: Put this logic in it's own little function. Makes reading the if a lot easier.

Comment (berry-langerak) [inline lib/scanners/base.js]: And then call it version_compare, and put it in it's own library. Or use an [existing version comparing library][1], I guess. ;)

[1]: https://www.npmjs.com/package/compare-version

File: lib/scanners/base.js [hunk 2]
```diff
@@ -34,17 +34,41 @@ var BaseScanner = function (beaconMapper, beaconRegionMapper, beaconRegionMonito
...
+                    hasPermissions();
+                    return;
+                }
+
+                Ti.Geolocation.requestLocationPermissions(permissionType, function(res) {
+                    if (res.success) {
+                        hasPermissions();
+                    }
+                });
+                return;
```
Comment (berry-langerak) [inline lib/scanners/base.js]: Isn't it possible to split the entire content of this `if` to a separate function? It's rather hard to read the control structure here, if you ask me.

File: lib/scanners/base.js [hunk 3]
```diff
@@ -34,17 +34,41 @@ var BaseScanner = function (beaconMapper, beaconRegionMapper, beaconRegionMonito
     this.init = function (networkIdentifier) {
         if (!_.isUndefined(networkIdentifier)) {
             self.networkId = networkIdentifier;
+            var hasPermissions = function () {
```
Comment (berry-langerak) [inline lib/scanners/base.js]: `hasPermissions` to me sounds like a function that returns whether or not something has permissions. You're not returning anything from this function, so it clearly does not do that. What I'm trying to say is that the naming seems to be off.

Comment (david4worx) [inline lib/scanners/base.js]: Great minds think alike ;)
Already explained the naming issue and rewrote the whole thing.

**Pull request merged**